src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -3436,6 +3463,35 @@ def update_auto_upgrade_profile(self, mc: ManagedCluster) -> ManagedCluster:
             mc.auto_upgrade_profile.node_os_upgrade_channel = node_os_upgrade_channel
         return mc
 
+    def update_guardrails_profile(self, mc: ManagedCluster) -> ManagedCluster:
```

Comment:
looks like there're many dup logic in update_guardrails_profile & set_up_guardrails_profile
possible to create a common function for those 2?

Reply:
refactored common code into helper funcion